From 256c5614313a37daf58117fdb3132eb7a641e032 Mon Sep 17 00:00:00 2001
From: De Huo <De.Huo@windriver.com>
Date: Tue, 6 Aug 2019 16:22:46 +0800
Subject: [PATCH] Add WRLinux1019 oval check for rule
 mount_option_krb_sec_remote_filesystems

Add WRLinux1019 oval check for rule mount_option_krb_sec_remote_filesystems.

Upstream-Status: Pending

Signed-off-by: De Huo <De.Huo@windriver.com>
---
 .../oval/wrlinux1019.xml                           | 40 ++++++++++++++++++++++
 1 file changed, 40 insertions(+)
 create mode 100644 linux_os/guide/services/nfs_and_rpc/nfs_configuring_clients/mounting_remote_filesystems/mount_option_krb_sec_remote_filesystems/oval/wrlinux1019.xml

diff --git a/linux_os/guide/services/nfs_and_rpc/nfs_configuring_clients/mounting_remote_filesystems/mount_option_krb_sec_remote_filesystems/oval/wrlinux1019.xml b/linux_os/guide/services/nfs_and_rpc/nfs_configuring_clients/mounting_remote_filesystems/mount_option_krb_sec_remote_filesystems/oval/wrlinux1019.xml
new file mode 100644
index 0000000..5bba154
--- /dev/null
+++ b/linux_os/guide/services/nfs_and_rpc/nfs_configuring_clients/mounting_remote_filesystems/mount_option_krb_sec_remote_filesystems/oval/wrlinux1019.xml
@@ -0,0 +1,40 @@
+<def-group>
+  <definition class="compliance" id="mount_option_krb_sec_remote_filesystems" version="1">
+    <metadata>
+      <title>Mount Remote Filesystems with Kerberos Security</title>
+      <affected family="unix">
+        <platform>multi_platform_wrlinux</platform>
+      </affected>
+      <description>The Kerberos security option should be enabled for all NFS mounts in /etc/fstab.</description>
+    </metadata>
+    <criteria operator="XOR">
+      <!-- these tests are designed to be mutually exclusive; either no nfs mounts exist in /etc/fstab -->
+      <!-- or all of the nfs mounts defined in /etc/fstab have the sec mount option specified -->
+      <criterion comment="remote nfs filesystems" test_ref="test_no_nfs_defined_etc_fstab_krb_sec" />
+      <criterion comment="remote nfs filesystems" test_ref="test_nfs_krb_sec_etc_fstab" />
+    </criteria>
+  </definition>
+  <ind:textfilecontent54_test check="all" check_existence="all_exist" comment="all nfs has krb_sec" id="test_nfs_krb_sec_etc_fstab" version="1">
+    <ind:object object_ref="object_nfs_krb_sec_etc_fstab" />
+    <ind:state state_ref="state_remote_filesystem_krb_sec" />
+  </ind:textfilecontent54_test>
+  <ind:textfilecontent54_object id="object_nfs_krb_sec_etc_fstab" version="1">
+    <ind:filepath>/etc/fstab</ind:filepath>
+    <ind:pattern operation="pattern match">^\s*\[?[\.\w-:]+\]?:[/\w-]+\s+[/\w-]+\s+nfs[4]?\s+(.*)$</ind:pattern>
+    <!-- the "not equal" operation essentially means all instances of the regexp -->
+    <ind:instance datatype="int" operation="not equal">0</ind:instance>
+  </ind:textfilecontent54_object>
+  <ind:textfilecontent54_state id="state_remote_filesystem_krb_sec" version="1">
+    <ind:subexpression operation="pattern match">^.*sec=krb5:krb5i:krb5p.*$</ind:subexpression>
+  </ind:textfilecontent54_state>
+  <ind:textfilecontent54_test check="all" check_existence="none_exist" comment="no nfs" id="test_no_nfs_defined_etc_fstab_krb_sec" version="1">
+    <!-- this test returns 'true' if /etc/fstab does not contain nfs/nfs4 mounts -->
+    <ind:object object_ref="object_no_nfs_defined_etc_fstab_krb_sec" />
+  </ind:textfilecontent54_test>
+  <ind:textfilecontent54_object id="object_no_nfs_defined_etc_fstab_krb_sec" version="1">
+    <ind:filepath>/etc/fstab</ind:filepath>
+    <ind:pattern operation="pattern match">^\s*\[?[\.\w-:]+\]?:[/\w-]+\s+[/\w-]+\s+nfs[4]?\s+.*$</ind:pattern>
+    <!-- the "not equal" operation below essentially means all instances of the regexp -->
+    <ind:instance datatype="int" operation="not equal">0</ind:instance>
+  </ind:textfilecontent54_object>
+</def-group>
-- 
1.9.1

